validation-test/stdlib/Algorithm.swift

```diff
@@ -24,15 +24,33 @@ public func == (
 }
 
 // FIXME(prext): move this struct to the point of use.
-Algorithm.test("min,max") {
-  expectEqual(2, min(3, 2))
-  expectEqual(3, min(3, 7, 5))
-  expectEqual(3, max(3, 2))
-  expectEqual(7, max(3, 7, 5))
-
-  // FIXME: add tests that check that min/max return the
-  // first element of the sequence (by reference equality) that satisfy the
-  // condition.
+Algorithm.test("Min/Max") {
+  expectEqual(0, min(0, 1))
+  expectEqual(0, min(1, 0))
+
+  expectEqual(0, min(0, 1, 1))
+  expectEqual(0, min(1, 0, 1))
+  expectEqual(0, min(1, 1, 0))
+
+  expectEqual(1, max(1, 0))
+  expectEqual(1, max(0, 1))
+
+  expectEqual(1, max(1, 0, 0))
+  expectEqual(1, max(0, 1, 0))
+  expectEqual(1, max(0, 0, 1))
+}
+
+Algorithm.test("Min/Max identity check") {
```

Comment:
You don't need to have separate value and identity tests.  All value tests could just check identity as well.